### PR TITLE
fix: dir package userConfigDir typo

### PR DESCRIPTION
**Pull request by JeyJeyGao**
1. fixed userConfigDir
2. fixed unit test on windows
3. added testdata for SysFS Open

Signed-off-by: Junjie Gao <junjiegao@microsoft.com>

### CODE REVIEW

**Review by shizhMSFT — APPROVED** [review]
LGTM

File: dir/path.go [hunk 1]
```diff
@@ -58,7 +58,7 @@ const (
 	TrustStoreDir = "truststore"
 )
 
-var userConfigDir = os.UserCacheDir // for unit test
+var userConfigDir = os.UserConfigDir // for unit test
```
Comment (patrickzheng200) [inline dir/path.go:61]: This is not for unit test right?

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Pull request merged**